README.md

```diff
@@ -1,24 +1,256 @@
+
+## Example Apps
+
+This repo comes with an [example app](app/) that is intended to showcase SDK functionality.
 Please add following line to your ```local.properties``` file to run example app
 
-```ABLY_KEY={YOUR ABLY API KEY}```
\ No newline at end of file
+- ```ABLY_KEY={YOUR ABLY API KEY}```
+
+## Usage
+
+This library is not yet published on any major publishing platform and only been used with an Android app. But you
+should be able to use it any Kotlin project by checking it out and adding it as a dependency module.
+
+You can clone library using
+```
+git clone https://github.com/ably-labs/GameRoomSDK.git
+```
+This will clone the multi module Gradle project including SDK and example app. The easiest way is to examine the
+example app to see how you can add SDK as a dependencey.
+
+If your project is in the same directory as ```ablygamesdk``` is add following to your ```settings.gradle``` file
+```
+include ':ablygamesdk'
+```
+Then in your app level build.gradle file add following
+
+```
+implementation project(path: ':ablygamesdk')
+```
+
+### AblyGame example usages
+
+Following are some examples of how to use functions that are provided by ```AblyGame```
+
+#### Game environment initialisation
+You will need to initialise a game environment in order to activate SDK functionality
+
+```kotlinlang
+val ablyGame = AblyGame.Builder(ABLY_API_KEY)
+            .scope(GAME_SCOPE)
+            .build()
+
+```
+You should replace ```ABLY_API_KEY``` with your own. Please login into [Ably dashboard](https://ably.com/login) to retrieve
+ your own api key.
+ ```GAME_SCOPE``` is coroutine scope that you intend to run your AblyGame instance in. A Typical application scope
+ for that matter could be defined as
+ ```
+ val applicationScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+ ```
+
+ #### Start a game
+
+```kotlinlang
+ablyGame.start {
+            when(it){
+                AblyGame.GameState.Started -> //TODO replace this with your action of game started
+                AblyGame.GameState.Stopped -> //TODO replace this with your action of game stopped
+            }
+        }
+```
+As you can see this method accepts a ```FlowCollector<GameState>``` for you to observe the changes on game state. You
+should make sure that the game is started before moving forward with other game functionalities
+
+#### Enter a game
+
+```
+someCoroutineScope.launch {
+            val enterResult = ablyGame.enter(gamePlayer)
+            if (enterResult.isSuccess) {
+               //enter successful
+            } else {
+               //enter failed
+            }
+        }
+```
+As you can see, you should invoke ```enter``` inside a coroutine scope.
+You must replace ```gamePlayer``` with your own instance of ```GamePlayer```
+
+#### Leave a game
+
+```kotlinlang
+someCoroutineScope.launch {
+            val leaveResult = MultiplayerGameApp.instance.ablyGame.leave(gamePlayer!!)
```

Comment:
Why here you've added the `MultiplayerGameApp.instance.` and also the `!!` at the end of the line? I'd say let's make this code example look like the rest of them :wink: 
```suggestion
            val leaveResult = ablyGame.leave(gamePlayer)
```